app/controllers/prior_authority/applications_controller.rb

```diff
@@ -65,7 +65,7 @@ def service
     }.freeze
 
     def order_and_paginate(query)
-      order_template = ORDERS.fetch(params[:sort_by], 'created_at ?')
+      order_template = ORDERS.fetch(params[:sort_by], 'created_at DESC')
```

Comment:
so this set a default sort direct when no sort by is passed in. This might be alright but would require deep knowledge of what params are passed when to be sure....

can I suggest

```
  def order_and_paginate(query)
      order_by = ORDERS[params[:sort_by]] ? params[:sort_by] : 'created_at'
      direction = if params[:sort_direction]
          params[:sort_direction] == 'ascending' ? 'ASC' : 'DESC'
        else
          order_by == 'created_at' ? 'DESC' : 'ASC'
        end        
      pagy(query.includes(:defendant).order(order_by => direction))
    end
```

NOTE this sets the default sort direction if none is passed in based on the field being sorted

NOTE: ORDERS hash is less important but could now be used to store default sort direction

Reply:
So I started off with something like that, but the issue I hit is that when sorting by client, we need to sort by both first name and last name, so `.order(column => direction)` doesn't work.

But you're right, my solution hinges on the fact that I know that the only situation where  `sort_by` is not provided is a situation where `sort_direction` is also not provided, and that's not made explicit in the code. So a more verbose working-out of `direction` is warranted. I'll sort that.